jujugui/static/gui/src/app/jujulib/bakery.js

```diff
@@ -73,14 +71,28 @@ var module = module;
       this._client = httpClient;
       this.storage = storage;
       this._onSuccess = params.onSuccess || (() => {});
-      this._noRedirects = !!params.noRedirects;
-      if (params.nonInteractive) {
-        this._visitPage = this._nonInteractiveVisitPage;
-      } else {
-        this._visitPage = params.visitPage || (visitURL => {
-          window.open(visitURL, 'Login');
-        });
-      }
+      this._visitPage = params.visitPage || (visitURL => {
+        window.open(visitURL, 'Login');
+      });
+      this._dischargeDisabled = false;
```

Comment:
Why not have this configurable in the constructor, then in `withoutDischarge` below you can set it as a value there.